[GTK][WPE] Some entities are missing brief descriptions, or are too long

siwei-li opened the pull request:
#### 39f864825530508e840e70c8510536e47e229e8c
<pre>
[GTK][WPE] Some entities are missing brief descriptions, or are too long
<a href="https://bugs.webkit.org/show_bug.cgi?id=242903">https://bugs.webkit.org/show_bug.cgi?id=242903</a>

Reviewed by Adrian Perez de Castro.

Fix the style of documentation by adding in missing brief descriptions
for the files listed below.
* Source/WebKit/Shared/API/glib/WebKitContextMenu.cpp:
* Source/WebKit/Shared/API/glib/WebKitContextMenuItem.cpp:
* Source/WebKit/Shared/API/glib/WebKitURIResponse.cpp:
* Source/WebKit/Shared/API/glib/WebKitUserMessage.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardListItem.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCredential.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFindController.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFormSubmissionRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitMemoryPressureSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitMimeInfo.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNavigationAction.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNetworkProxySettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNotification.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitPlugin.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitPolicyDecision.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitResponsePolicyDecision.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitScriptDialog.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURIUtilities.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitVersion.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteData.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsitePolicies.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitInputMethodContextGtk.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:

Canonical link: <a href="https://commits.webkit.org/253104@main">https://commits.webkit.org/253104@main</a>
</pre>

aperezdc commented on Source/WebKit/Shared/API/glib/WebKitContextMenu.cpp:
```diff
@@ -172,6 +176,8 @@ void webkit_context_menu_append(WebKitContextMenu* menu, WebKitContextMenuItem*
  *
  * Inserts @item into the @menu at the given position.
+ *
+ * Inserts @item into the @menu at the given position.
```
In this case the first phrase of the long description is the same as the brief description, so we don't need to repeat it. The documentation generator writes the brief one as part of the detailed output for each element, you can see this e.g. in the [page for WebKitWebView](https://people.igalia.com/aperez/Documentation/webkit2gtk-4.1/class.WebView.html).

In this kind of situations we only need to add the empty line to convert the first phrase into the brief description:

```
 * [...]
 * Inserts @item into the @menu at the given position.
 *
 * If @position is negative, or is larger then the number of items
 * in the #WebKitContextMenu, the item is added on to the end of
 * [...]
```

aperezdc commented on Source/WebKit/Shared/API/glib/WebKitContextMenu.cpp:
```diff
@@ -192,6 +198,8 @@ void webkit_context_menu_insert(WebKitContextMenu* menu, WebKitContextMenuItem*
  *
  * Moves @item to the given position in the @menu.
+ *
+ * Moves @item to the given position in the @menu.
```
Same comment here. There are a few more cases like this one in the patch, so I won't comment on all of them&mdash;sorry about making you change all those 😇

aperezdc commented on Source/WebKit/Shared/API/glib/WebKitUserMessage.cpp:
```diff
@@ -291,6 +291,8 @@ GUnixFDList* webkit_user_message_get_fd_list(WebKitUserMessage* message)
  * @message: a #WebKitUserMessage
  * @reply: a #WebKitUserMessage to send as reply
  *
+ * Send a reply to @message.
```
Some of those would be nice to write without `@var` and `#type` markup. For example in this particular case I think something like this would be neater:

```suggestion
 * Send a reply to an user message.
```

Anyway, we can keep the focus of this patch in having brief descriptions, and leave actual content edits for a separate patch. In many cases one would need to know an understanding of how the API functions work, which is something that we can't expect from new contributors 😉&mdash;so I am mostly writing this down to help ourselves remember later about this.

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
```diff
@@ -174,6 +174,8 @@ WebKitCookieManager* webkitCookieManagerCreate(WebKitWebsiteDataManager* dataMan
  * @filename: the filename to read to/write from
  * @storage: a #WebKitCookiePersistentStorage
  *
+ * Set non-session cookies.
```
For example this brief description is excellent: short and to the point, with all the details about how it works in the longer text below 👍🏼

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.cpp:
```diff
@@ -102,7 +104,7 @@ G_DEFINE_BOXED_TYPE(WebKitGeolocationPosition, webkit_geolocation_position, webk
  * @longitude: a valid longitude in degrees
  * @accuracy: accuracy of location in meters
  *
- * Create a new #WebKitGeolocationPosition
+ * Create a new #WebKitGeolocationPosition.
```
Nice catch, I hadn't realized before that some brief descriptions were missing the dot at the end!

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.cpp:
```diff
@@ -223,8 +233,10 @@ guint16 webkit_security_origin_get_port(WebKitSecurityOrigin* origin)
  * webkit_security_origin_is_opaque:
  * @origin: a #WebKitSecurityOrigin
  *
+ * This function returns %FALSE.  
+ *
  * This function returns %FALSE. #WebKitSecurityOrigin is now a simple
- * wrapper around a &lt;protocol, host, port&gt; triplet, and no longer
+ * wrapper around a <protocol, host, port> triplet, and no longer
```
Good catch as well, no need to use SGML entities (the `&stuff;` things) with `gi-docgen`. This makes the documentation easier to read and edit when making changes to the code.

aperezdc reviewed (changes requested):
Thanks for working on this, we definitely want to improve how the documentation reads and looks.

This is going in the right direction, but please take a look at my comment about trying to avoid repeating the same phrase for both the brief description of an element and the first phrase of its long description. With that solved, I think we could then merge the patch.

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
```diff
@@ -445,6 +456,8 @@ gboolean webkit_authentication_request_is_retry(WebKitAuthenticationRequest* req
  * @request: a #WebKitAuthenticationRequest
  * @credential: (transfer none) (allow-none): A #WebKitCredential, or %NULL
  *
+ * Authenticate the #WebKitAuthenticationRequest.  
```
This line introduces trailing whitespace, let's remove it.

```suggestion
 * Authenticate the #WebKitAuthenticationRequest.
```

siwei-li replied:
Ah thank you for spotting these out! I was just doing some simple grep for the trailing space pattern; must have missed some of them. Will comb through these again

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
```diff
@@ -204,6 +206,7 @@ void webkit_cookie_manager_set_persistent_storage(WebKitCookieManager* manager,
  * @policy: a #WebKitCookieAcceptPolicy
  *
  * Set the cookie acceptance policy of @cookie_manager as @policy.
+ * 
```
Another trailing space here:

```suggestion
 *
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitFindController.cpp:
```diff
@@ -474,6 +484,8 @@ void webkit_find_controller_search_previous(WebKitFindController* findController
  * @find_options: a bitmask with the #WebKitFindOptions used in the search
  * @max_match_count: the maximum number of matches allowed in the search
  *
+ * Counts the number of matches for @search_text 
```
Missing dot here, and trailing space:

```suggestion
 * Counts the number of matches for @search_text.
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp:
```diff
@@ -71,8 +72,10 @@ WebKitJavascriptResult* webkit_javascript_result_ref(WebKitJavascriptResult* jav
- * reference count drops to 0, all memory allocated by the #WebKitJavascriptResult is
+ * Atomically decrements the reference count of @js_result by one.
+ *
+ * If the reference count drops to 0, 
```
Trailing space:

```suggestion
 * If the reference count drops to 0,
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.cpp:
```diff
@@ -223,8 +231,10 @@ guint16 webkit_security_origin_get_port(WebKitSecurityOrigin* origin)
  * webkit_security_origin_is_opaque:
  * @origin: a #WebKitSecurityOrigin
  *
+ * This function returns %FALSE.  
```
Two more unneeded trailing spaces here:

```suggestion
 * This function returns %FALSE.
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
```diff
@@ -979,6 +989,7 @@ void webkit_web_context_clear_cache(WebKitWebContext* context)
  * @proxy_settings: (allow-none): a #WebKitNetworkProxySettings, or %NULL
  *
  * Set the network proxy settings to be used by connections started in @context.
+ * 
```
Trailing space:

```suggestion
 *
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
```diff
@@ -1011,9 +1022,10 @@ static DownloadsMap& downloadsMap()
- * starting a download from a particular #WebKitWebView use
+ * Requests downloading of the specified URI string.
+ *
+ * The download operation will not be associated to any #WebKitWebView, 
```
Trailing space:

```suggestion
 * The download operation will not be associated to any #WebKitWebView,
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
```diff
@@ -1812,6 +1849,7 @@ void webkit_web_context_initialize_notification_permissions(WebKitWebContext* co
  * @message: a #WebKitUserMessage
  *
  * Send @message to all #WebKitWebExtension<!-- -->s associated to @context.
+ * 
```
Trailing space:

```suggestion
 *
```

aperezdc commented on Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
```diff
@@ -357,6 +361,7 @@ GtkWidget* webkit_web_view_new_with_context(WebKitWebContext* context)
  * @web_view: the related #WebKitWebView
  *
  * Creates a new #WebKitWebView sharing the same web process with @web_view.
+ * 
```
Trailing space:

```suggestion
 *
```

aperezdc commented on Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
```diff
@@ -406,6 +412,7 @@ GtkWidget* webkit_web_view_new_with_settings(WebKitSettings* settings)
  * @user_content_manager: a #WebKitUserContentManager.
  *
  * Creates a new #WebKitWebView with the given #WebKitUserContentManager.
+ * 
```
One more here:

```suggestion
 *
```

aperezdc reviewed (changes requested):
@siwei-li Thanks for updating this patch, we're almost there!

We try to avoid introducing unneeded whitespace accidentally in patches, and I have noticed that there are a few in your patch. Could you take a look? After that this will be ready for merging 😺

aperezdc reviewed (approved):
This is now ready to go. Thanks @siwei-li for your patience with all the review steps 💯

siwei-li committed:
[GTK][WPE] Some entities are missing brief descriptions, or are too long
https://bugs.webkit.org/show_bug.cgi?id=242903

Reviewed by Adrian Perez de Castro.

Fix the style of documentation by adding in missing brief descriptions
for the files listed below.
* Source/WebKit/Shared/API/glib/WebKitContextMenu.cpp:
* Source/WebKit/Shared/API/glib/WebKitContextMenuItem.cpp:
* Source/WebKit/Shared/API/glib/WebKitURIResponse.cpp:
* Source/WebKit/Shared/API/glib/WebKitUserMessage.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardListItem.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCredential.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFindController.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFormSubmissionRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitMemoryPressureSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitMimeInfo.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNavigationAction.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNetworkProxySettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNotification.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitPlugin.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitPolicyDecision.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitResponsePolicyDecision.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitScriptDialog.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURIUtilities.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitVersion.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteData.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebsitePolicies.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitInputMethodContextGtk.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:

Canonical link: https://commits.webkit.org/253104@main
